src/copy.c

```diff
@@ -233,11 +242,18 @@ timescaledb_CopyFrom(CopyChunkState *ccstate, List *range_table, Hypertable *ht)
 	bistate = GetBulkInsertState();
 	econtext = GetPerTupleExprContext(estate);
 
-	/* Set up callback to identify error line number */
-	errcallback.callback = CopyFromErrorCallback;
-	errcallback.arg = (void *) ccstate->fromctx.cstate;
-	errcallback.previous = error_context_stack;
-	error_context_stack = &errcallback;
+	/* Set up callback to identify error line number.
+	 *
+	 * It is not necessary to add an entry to the error context stack if we do
+	 * not have a CopyState. In that case, we just use the existing error
+	 * already on the context stack. */
+	if (ccstate->context_type == CCS_CSTATE)
```

Comment:
Do we need to save the `context_type` in the state? I cannot see it being used anywhere.

Reply:
Not sure I follow: isn't this the location where it is used?

Reply:
Since it seemed to be a little unclear, I removed the context_type and instead have two fields: one with the heap scan descriptor and one with the copy state. This will make it type-safe(ish?) and avoid the risk of casting between the two types accidentally.